src/nupic/ntypes/ArrayBase.cpp

```diff
@@ -277,27 +277,27 @@ bool ArrayBase::has_buffer() const { return (buffer_.get() != nullptr); }
  * args:
  *    a         - Destination buffer
  *    offset    - Index used as starting index. (defaults to 0)
- *    maxsize   - Total size of destination buffer (if 0, use source capacity)
+ *    maxsize   - Total size of destination buffer (if 0, use source size)
  *                This is used to allocate destination buffer size (in counts).
  */
 void ArrayBase::convertInto(ArrayBase &a, size_t offset, size_t maxsize) const {
   if (maxsize == 0)
-    maxsize = getMaxElementsCount() + offset;
+    maxsize = getCount() + offset;
```

Comment:
can this not change? The original capacity/MaxElementCount() seemed more fit?

Reply:
The problem I was trying to solve by changing this was that in a Fan-In condition I need to set the buffer size to the combined size of all of the buffers that will contribute.  That is the maxsize that is being passed in.  But if it is not a Fan-In condition (maxsize == 0) and I need to allocate a buffer then I should only allocate what I need, not the capacity of the source buffer.